src/http_microhttpd.c

```diff
@@ -552,7 +566,7 @@ static int preauthenticated(struct MHD_Connection *connection,
 
 		if (!try_to_authenticate(connection, client, host, url)) {
 			/* user used an invalid token, redirect to splashpage but hold query "redir" intact */
-			return encode_and_redirect_to_splashpage(connection, redirect_url);
+			return encode_and_redirect_to_splashpage(connection, redirect_url, querystr);
```

Comment:
querystr will be NULL here. Won't that be a problem?

Reply:
Not a problem and is intended. This bounces the client back to the beginning if the token is expired - just as if the CPD had triggered the splash page (be it NDS splash or FAS splash).